### ISSUE TITLE
Duplicate cell contents when using functional components with ag-grid-react 23

### ISSUE THREAD

**Issue by JosNun** [ag-grid/ag-grid#3731]
**I'm submitting a ...** 
```
[x] bug report => see 'Providing a Reproducible Scenario'
[] feature request => do not use Github for feature requests, see 'Customers of ag-Grid'
[] support request => see 'Requesting Community Support'
```

**Providing a Reproducible Scenario**

[Code Sandbox Reproduction](https://codesandbox.io/s/ag-grid-react-2300-double-cell-renderer-bug-pxcp1)

**Current behavior**

When using functional components in ag-grid-react (v23.0.0), it can sometimes render multiple components in the same cell when scrolling quickly.

*Edit*
Upon further investigation, it appears as if this is an issue when reactComponent.js tries to render the component as static markup. If I add `disableStaticMarkup` to the grid component, it restores the old behavior.

**Expected behavior**

A single component will render in each cell.

*Temporary Workarounds*
To restore the old functionality, adding `disableStaticMarkup` to the grid component seems to work. 

If you must use static markup for some reason (maybe potential performance boost?), you can add the following to your css to `display: none` the second component. Although it still renders, it should not affect the cell's layout.

```css
.ag-react-container > :not(:first-child) {
    display: none !important;
}
```

**Please tell us about your environment:**

Windows
React 16.12

**ag-Grid version:** 23

ag-grid-community: 23.0.1
ag-grid-react: 23.0.0

**Browser:**

  - Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/80.0.3987.132 Safari/537.36
  - Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:74.0) Gecko/20100101 Firefox/74.0

**Language:** all

**Comment by rmc-software** [comment]
Thanks for raising this with us. We have created an internal jira to track this: AG-4020.

See our pipeline for progress updates.

https://www.ag-grid.com/ag-grid-pipeline/

A fix is scheduled for our next release v23.1 due in 4/5 weeks.

**Comment by ydostyle** [comment]
@rmc-software Didn't fix the problem in v23.1

**Comment by Latchy** [comment]
I am also experiencing this issue, in version 24.1
The first row is rendering the component twice. 
Using the API to force a rerender when the Grid has loaded fixes the issue, but not without a flash of incorrectly styled rows.
Can you please reopen this issue and find a fix?

Thanks

**Comment by AGrab** [comment]
> 
> 
> I am also experiencing this issue, in version 24.1
> The first row is rendering the component twice.
> Using the API to force a rerender when the Grid has loaded fixes the issue, but not without a flash of incorrectly styled rows.
> Can you please reopen this issue and find a fix?
> 
> Thanks

I have the same issue. After updating ag-grid community and enterprise from 23.2.1 to 24.1.0 the content of my cell renderer is being duplicated in the first row.

**Comment by rmc-software** [comment]
Thanks for reporting this. I have reopened this issue.

**Comment by kiril-matev** [comment]
Hello!

Thank you all for reporting this. 

We've now reproduced this behavior as described in v24.1.

We're tracking this bug with the following item:
AG-4180 [React] React function component cell renderer briefly shows two copies of the cell value in a cell

We try to fix bugs from one release to the next, so this should be fixed in the next release or the one after if it was raised too close to the next release date.

You can check whether this item is selected for implementation by taking a look at the NEXT RELEASE tab of the product pipeline:
https://www.ag-grid.com/ag-grid-pipeline/

I would recommend signing up for automatic release notifications for ag-Grid. Then, following a release, check whether the item was implemented by looking for it in the changelog page:
https://www.ag-grid.com/ag-grid-changelog/

Thanks again!

**Comment by kiril-matev** [comment]
Hello!

I just wanted to follow up with an update on one of the two issues reported here.

We have just shipped a new version of ag-Grid - v24.1.1, which resolves this issue:
**AG-4712 [Regression] React functional component is shown twice in the first row until scrolled out of view**

Please get this latest version to have issue AG-4712 resolved.

I want to confirm that the second issue is still outstanding in our backlog to fix:
**AG-4180 [React] React function component cell renderer briefly shows two copies of the cell value in a cell**

Thank you for following this thread.

**Comment by makinggoodsoftware** [comment]
Hi,

In order to help us keep our issues queue clear, we are going to close this issue

The official way to check the status for an issue raised here is to check its AG-xxxx ID against our pipeline and changelog

…the pipeline (if not yet developed)

https://www.ag-grid.com/ag-grid-pipeline/

... or the changelog, (if the issue has been resolved)

https://www.ag-grid.com/ag-grid-changelog/

We keep issues raised in the pipeline through github open for a while so that the community can add some feedback to them, and close them when there is no activity for a while.

If you want to add further feedback to a closed ticket with this comment, we would recommend opening a new ticket as we don’t monitor closed issues.
